packages/eslint-plugin-udemy/helpers/is-udlite-file.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+let cachedUDLiteDirs = null;
+
+exports.isUDLiteFile = filename => {
+    if (!filename.includes('/static/src/udemy/js/')) {
+        return false;
+    }
+
+    if (cachedUDLiteDirs === null) {
+        const jsDir = path.join(filename.split('/static/src/udemy/js/')[0], 'static/src/udemy/js');
+        cachedUDLiteDirs = findUDLiteDirs(jsDir);
+    }
+
+    return filename.includes('udlite') || cachedUDLiteDirs.some(dir => filename.startsWith(dir));
+};
+
+function findUDLiteDirs(dir) {
+    if (!fs.existsSync(dir)) {
+        return [];
+    }
+
+    const udliteDirs = [];
+    if (
+        fs.existsSync(path.join(dir, 'udlite-app.js')) &&
+        !fs.existsSync(path.join(dir, 'app.js'))
+    ) {
+        udliteDirs.push(dir);
+    } else if (fs.existsSync(path.join(dir, 'udlite.md'))) {
+        udliteDirs.push(dir);
```

Comment:
Would it be faster to walk the directories recursively once and then just do array / set includes / contains? Rather than a bunch of existsSync calls? Or if it's negligible don't bother 😊

Reply:
Done; thanks! That makes perfect sense, as we're walking the directories anyway after doing these checks.